docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,188 @@
+- how Corvus.Tenancy and Marain.Tenancy help with building multi-tenanted services.
+
+## What does multi-tenanted mean? 
+Let’s start by thinking about the distinction between *multi-tenant* and *multi-user*.
+
+The chief problem with giving multiple users access to a computer is that we need a fair way of ensuring that they don’t tread on each other’s toes.
+
+Initially, people used to book time on the computer. They would get exclusive access to a giant room of glowing valves, for a specific period of time. Then get physically booted out.
+
+By the 1970s, the slightly confusingly named [time sharing](https://en.wikipedia.org/wiki/Time-sharing) model predominated. Multiple users could access a single computer at the same time, sharing the resources.
+
+The security and fair-shares implications of this ultimately led to the development of isolated process models, virtual machines, containers, and the cloud/edge computing model we know today. 
+
+As part of this shift, the line between “computer user” and “solution user” has also blurred.
+
+In a modern web application, for example, there is frequently a single logical “solution” serving hundreds, thousands, or millions of concurrent users.
+
+That solution is likely implemented using many compute and storage resources, distributed across various physical and virtual machines, but it is conceptually *one solution, lots of users*. To developer and user alike, it looks like a single set of resources, serving many users. Those users may work together in logical groups (like teams, or organizations) but they are fundamentally sharing a single solution.
+
+A multi-tenanted solution, on the other hand, can be thought of as *lots of solutions, lots of users*. To the *owner* of the multi-tenanted solution, it looks conceptually like *many isolated solutions*, each of which independently serves many users. To any individual tenant, their “tenancy” looks like a single set of resources serving their users, and no-one else’s.
+
+[multi-user v. multi tenant](multi-user v multi-tenant.png)
+
+## When should a solution be multi-tenanted, not multi-user?
+A multi-tenanted solution is required when one or more groups of users have a strong requirement for *isolation* from other groups of users.
+
+Typically this isolation will manifest in one or more of the following ways:
+1. The ability for a tenant to provision, manage, and monitor a collection of resources independently of other tenants.
+1. The ability for a tenant to bring their own resources (including, for example, application code, configuration, data, cryptographic keys, policy) and integrate into their solution
+1. The ability to provide particular kinds of security boundary around their solution and control identity and access (including the ability to deny access to the entity that provides the tenant with those resources)
+1. The inability for one tenant to determine the existence or otherwise of another tenant.
```

Comment:
Is there also something more specific than "knowing" around resource / performance demand isolation, i.e. "noisy neighbour" pattern?

Reply:
I had intended this to be a special part of "knowing" (I was thinking "sensing") but I think it is too important not to give it its own bullet point. So I have done.